### PR TITLE
new_with_info construtor for encoder

### PR DESCRIPTION
**Pull request by DeniseBryson**
Hi,
I added a new constructor for the encoder. You can pass the info struct to create a new encoder. With that you can decode an image change something and encode it again more easily.
I also added an example to shows how to use the function, called change-png-info.rs .
Thanks to @hoijui for the help.

### CODE REVIEW

File: examples/change-png-info.rs [hunk 1]
```diff
@@ -0,0 +1,53 @@
+// SPDX-FileCopyrightText: 2023 Lukas Schattenhofer <lukas.schattenhofer@ose-germany.de>
+// SPDX-License-Identifier: MIT OR Apache-2.0
```
Comment (fintelia) [inline examples/change-png-info.rs:2]: Please remove. We track authorship via git history and all source code in the repository is MIT OR Apache-2.0 as per the LICENSE files in the root directory

File: src/encoder.rs [hunk 2]
```diff
@@ -169,6 +169,14 @@ impl<'a, W: Write> Encoder<'a, W> {
         }
     }
 
+    pub fn new_with_info(w: W, info: Info<'a>) -> Encoder<'a, W> {
```
Comment (fintelia) [inline src/encoder.rs:172]: Please name this `with_info` to be consistent with the "General constructors" entry in the [API guidelines](https://rust-lang.github.io/api-guidelines/naming.html#casing-conforms-to-rfc-430-c-case).

**Review by fintelia — COMMENTED** [review]
Thanks for this PR! This will make it much easier to re-encode with the same metadata as an existing image

I see the CI is failing due to a formatting issue. You should be able to fix that by running `cargo fmt`

**Comment by DeniseBryson** [comment]
Thanks for the comments, should be all fixed now.

**Pull request merged**